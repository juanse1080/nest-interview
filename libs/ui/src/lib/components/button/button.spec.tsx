import { composeStories } from '@storybook/testing-react';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';

import userEvent from '@testing-library/user-event';
import Button from './button';
import defaultStory, * as stories from './button.stories';

const composedStories = composeStories(stories);

describe(`[${defaultStory.title}]`, () => {
  Object.entries(composedStories).forEach(([story, Component]) => {
    it(`should render ${story}`, () => {
      const { asFragment } = render(<Component />);
      expect(asFragment).toMatchSnapshot();
    });
  });
});

describe('Button', () => {
  it('Content', async () => {
    const name = 'hello world';

    render(<Button>{name}</Button>);
    const button = screen.getByRole('button', { name });

    expect(button).toHaveTextContent(name);
  });

  it('Click button', async () => {
    const name = 'hello world';
    const user = userEvent.setup();

    const onClick = jest.fn();
    render(<Button onClick={onClick}>{name}</Button>);

    const button = screen.getByRole('button', { name });
    await user.click(button);

    expect(onClick).toHaveBeenCalled();
  });

  it('Disabled button', async () => {
    const name = 'hello world';
    const user = userEvent.setup();

    const onClick = jest.fn();
    render(
      <Button onClick={onClick} disabled>
        {name}
      </Button>
    );

    const button = screen.getByRole('button', { name });
    await user.click(button);

    expect(onClick).not.toHaveBeenCalled();
  });
});
