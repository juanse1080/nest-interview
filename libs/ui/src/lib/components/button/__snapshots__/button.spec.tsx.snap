// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Button] should render Basic 1`] = `[Function]`;

exports[`[Button] should render Disabled 1`] = `[Function]`;
